modules/SDL.lua

```diff
@@ -1,15 +1,67 @@
 require('os')
 local sdl_logger = require('sdl_logger')
 local config = require('config')
+local console = require('console')
 local SDL = { }
 
 require('atf.util')
 
+SDL.buildOptions = {}
 SDL.exitOnCrash = true
 SDL.STOPPED = 0
 SDL.RUNNING = 1
 SDL.CRASH = -1
 
+local usedBuildOptions = {
+  remoteControl =  {
+    sdlBuildParameter = "REMOTE_CONTROL",
+    defaultValue = "ON"
+  },
+  extendedPolicy =  {
+    sdlBuildParameter = "EXTENDED_POLICY",
+    defaultValue = "PROPRIETARY"
+  }
+}
+
+local function readParameterFromCMakeCacheFile(paramName)
+  local pathToFile = config.pathToSDL .. "/CMakeCache.txt"
+  if is_file_exists(pathToFile) then
+    local paramValue, paramType
+    for line in io.lines(pathToFile) do
+      paramType, paramValue = string.match(line, "^%s*" .. paramName .. ":(.+)=(%S*)")
+      if paramValue then
+        return paramValue, paramType
+      end
+    end
+  end
+  return nil
+end
+
+local function setSdlBuildOption(optionName, sdlBuildParam, defaultValue)
+  local value, paramType = readParameterFromCMakeCacheFile(sdlBuildParam)
+  if value == nil then
+    value = defaultValue
+    local msg = "SDL build option " ..
+      sdlBuildParam .. " is unavailable.\nAssume that SDL was built with " ..
+      sdlBuildParam .. " = " .. defaultValue
+    print(console.setattr(msg, "cyan", 1))
+  else
+    if paramType == "UNINITIALIZED" then
+      value = nil
+      local msg = "SDL build option " ..
+        sdlBuildParam .. " is unsupported."
+      print(console.setattr(msg, "cyan", 1))
+    end
+  end
+  SDL.buildOptions[optionName] = value
```

Comment:
@aderiabin result of this method is a modification of SDL table. It is very not obvious. You could make this function as member SDL.

Reply:
@LuxoftAKutsan  Description in LDoc format was added (see 5aec3b3)
SDL module as parameter for local function was added (see b5835e5)